src/java/org/apache/cassandra/db/partitions/BTreePartitionUpdate.java

```diff
@@ -145,6 +149,19 @@ public static BTreePartitionUpdate singleRowUpdate(TableMetadata metadata, Decor
         return singleRowUpdate(metadata, key, row.isStatic() ? null : row, row.isStatic() ? row : null);
     }
 
+    @Override
+    public PartitionUpdate withOnlyPresentColumns()
+    {
+        Set<ColumnMetadata> columnSet = new HashSet<>();
+
+        for (Row row : rows())
+            for (ColumnData column : row)
+                columnSet.add(column.column());
+
+        RegularAndStaticColumns columns = RegularAndStaticColumns.builder().addAll(columnSet).build();
+        return new BTreePartitionUpdate(metadata, partitionKey, holder.withColumns(columns), deletionInfo.mutableCopy(), false);
+    }
+
```

Comment:
For Review:
`withOnlyPresentColumns `, `accumulatedDataSize` and `unsharedHeapSize` are coming from 5.0 before `PartitionUpdate` became an interface. 

The implementations have moved to `BTreePartitionUpdate` and `TriePartitionUpdate`.